pyglmnet/pyglmnet.py

```diff
@@ -692,8 +695,20 @@ def fit(self, X, y):
             raise ValueError('Input data should be of type ndarray (got %s).'
                              % type(X))
 
-        n_features = X.shape[1]
+        if not X.ndim == 2:
+            raise ValueError('Input data should be of shape' +
+                             '(n_observations, n_features)')
+
+        y = np.asarray(y)
+        if y.ndim == 2:
```

Comment:
why not raise an error for y the same way as for X?

Reply:
As it stands the code allows both y to be of shape (obs,) or (obs, 1).

Reply:
hmm, that's confusing indeed. I think historically it was allowed because we had a multinomial distribution but now that's removed it shouldn't be the case.

Reply:
Only allowing 1D arrays for y now, and validating against this.